rmgpy/molecule/molecule.py

```diff
@@ -1866,12 +1867,34 @@ def kekulize(self):
         """
         kekulize(self)
 
-    def assignAtomIDs(self):
+    def assignAtomIDs(self, start=-1):
         """
-        Assigns an ID number to every atom in the molecule for tracking purposes.
+        Assigns an index to every atom in the molecule for tracking purposes.
+        If start is not specified, the system will use random integers for each index
+        Otherwise the system will start at `start` and increment one each time.
         """
+
+        from random import randint
         for i, atom in enumerate(self.atoms):
-            atom.id = i
+            if start == -1:
+                # use entire range of integers to label atoms
+                atom.id = randint(-2**15,2**15)
```

Comment:
Good question. Generating IDs from a starting value (like previously going 1 to N) is almost certainly going to cause clashes of atom IDs after two labeled atoms are merged in generateReactions. Generating IDs over the entire range of integer values is much more likely to reduce the clashing IDs when merging two molecules (which is what random IDs seeks to accomplish). When a clash occurs, we would need to reassign IDs properly for all resonance structures (possibly remaking resonance structures which is quite time intensive)

One alternative to completely random numbers could be to label starting from -2^-15 till 2^15 and then loop back over after labeling ~65,000 atoms (not resetting after each call). If you know that this would reduce conflicts and/or computational time, let me know.

The atomIDs cannot be set post-merging since this can result in incorrect degeneracy values when trying to generate reactions when multiple resonance structures are present. Lets say you have two reactants reacting together, and each has multiple resonance structures. In generateReactions, all combinations of resonance structures will be merged and the possible reactions output. Then `findDegeneracies` would use the atomIDs for all the sets of products to find the degeneracy. If different resonance structures are labeled differently (which can happen when relabeling after the merging of structures), the degeneracy could be higher than expected.